src/vcpkg/versionedsolver.cpp

```diff
@@ -0,0 +1,673 @@
+#include <vcpkg/base/messages.h>
+#include <vcpkg/base/strings.h>
+
+#include <vcpkg/cmakevars.h>
+#include <vcpkg/dependencies.h>
+#include <vcpkg/packagespec.h>
+#include <vcpkg/portfileprovider.h>
+#include <vcpkg/sourceparagraph.h>
+#include <vcpkg/triplet.h>
+
+namespace
+{
+    using namespace vcpkg;
+
+    /**
```

Comment:
I'm somewhat opposed to adding a separate .cpp when the header is still the same. I'm not seeing why being in a separate file like this clarifies things.

(That is, I have a mild preference to go back into dependencies.cpp, but I would be OK with the header functionality split into a new header to match the separate cpp)

Reply:
There are a lot of similar concepts between the two algorithms, but nothing is actually shared. This makes it more difficult to work on since searches in the file can often find hits in the other algorithm.

Reply:
I do have a long-term desire to unify the two algorithms somehow, so I'd also be OK with merging it back.

Reply:
I've merged them back together.